### DISCUSSION TITLE
Undo last CLI import? Tons of duplicates.

### DISCUSSION THREAD

**Discussion by adambeck7** [immich-app/immich#3283] [Q&A] [answered]
I ran the CLI import, it failed part way through when it hit a .mov file it couldn't import. I ran it again and it appears to have imported all the faces and all the previously imported immich photos again. Is there a way I can remove everything imported in the last 24hours? Or do I have to delete and start the import from scratch?

**Comment by alextran1502** [comment]
The CLI won't be uploading/importing files that are already on the server. If you read the message, the actual file count that would be processed is different

**Reply by alextran1502** [reply]
And there is no undo mechanism

**Reply by adambeck7** [reply]
Yea, the previously indexed vs how many it had to add was different. I think I might've messed something up. I have `/mnt/user/Pictures` bound to the `/usr/src/app/existing_pictures` and` /usr/src/app/upload` is bound to `/mnt/user/Pictures/Immich/`. I'm guessing since the upload is nested within the existing pictures, could that have caused a bunch of dupes to have been uploaded? Including all the faces that had already been indexed? 

Is there a way to just delete everything I uploaded via the CLI yesterday? I could just run a raw SQL query if I know what I'm targeting, but I don't want to mess it up.

**Reply by alextran1502** [reply, marked as answer]
You can follow this guide to purge data from Immich https://immich.app/docs/FAQ#how-can-i-purge-data-from-immich

I am not sure how you are setting up your `.env` but maybe use the recommended value first and see if it works for you, and then customize it incrementally to fit your needs.

**Reply by adambeck7** [reply]
Thanks for that. I had it set up as recommended for months, it's been great. Then I had to mess with a good thing. Sorry about the trouble and thanks for the help!

**Comment by ngxson** [comment]
Maybe late but I kinda found a solution: To use `/api/asset` to list all assets stored in my account, then sort by `asset.updatedAt`

Here is the code in javascript. You can add a `.filter` to get only assets updated in the last X minutes, then call the delete assets endpoint to delete them (https://documentation.immich.app/docs/api/delete-asset)

```javascript
const getTime = str => (new Date(str)).getTime();

(async function () {
  const API_KEY = '..........................';
  const res = await fetch('https://YOUR_IMMICH_HOST_HERE.com/api/asset', {
    headers: { 'x-api-key': API_KEY }
  })
  const assets = await res.json();

  assets.sort((a, b) => getTime(b.updatedAt) - getTime(a.updatedAt));

  for (let i = 0; i < 20; i++) {
    console.log(assets[i].updatedAt, assets[i].originalPath);
  }
})();

```

**Reply by ngxson** [reply]
Btw @alextran1502 Can we have `sortBy` and `limit` parameters when calling `getAllAssets`? Knowing that in the documentation, we already have `skip` parameter but it doesn't seem to be used.

**Reply by adambeck7** [reply]
Thanks! Since mine was on a single day, it was easier for me to just run a query that deleted everything from public.assets that had a fileCreatedDate of `date`. Not sure if the delete cascaded and deleted everything associated with those assets or not, but it fixed my issue.
